### DISCUSSION TITLE
Records management via API

### DISCUSSION THREAD

**Discussion by tofazzz** [coredns/coredns#4737] [Ideas]
Howdy,

I'm currently using PowerDNS as a backend for a GSLB solution that I'm developing in Go, but I'm looking to switch to CoreDNS and I wanted to ask if there is any way to manage records via API in real time instead of, i.e manipulate the host file?

In PowerDNS there is the utility pdnsutil which you can execute to feed in the record parameters with the flags create, update or delete and it feeds it directly into the zone without any reload:

pdnsutil add-record <domain> <record> <type> <ttl> <ip address>
pdnsutil delete-rrset <domain> <record> <type>

If not currently present, is it something that you have in the roadmap or in case on a post-it as an idea? :)

Thank you for your great job in developing CoreDNS!

**Comment by miekg** [comment]
Hi

No an API in coredns itself will not happen. The design behind powerdns is a sane one,
simple API to the database, database takes care of authn/authz and provisioning.

There are some database plugins available, but there isn't one in coredns itself (yet).

**Reply by tofazzz** [reply]
Thank you for your confirmation!
